src/libiotc/iotc_roughtime_client.cc

```diff
@@ -0,0 +1,225 @@
+#define WOLFSSL_LEANPSK // 20190124
+// #ifndef __linux
+// #define __linux // 20190124
+// #endif
+
+#include <iotc_bsp_io_net.h>
+
+#include <arpa/inet.h>
+#include <assert.h>
+#include <errno.h>
+#include <inttypes.h>
+#include <netdb.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "client.h"
+#include "clock.h"
+#include "protocol.h"
+#include <iotc_bsp_rng.h>
+#include <iotc_bsp_time.h>
+#include <iotc_debug.h>
+#include <iotc_roughtime_client.h>
+#include <wolfssl/options.h>
+#include <wolfssl/ssl.h>
+#include <wolfssl/wolfcrypt/settings.h>
+
+// kTimeoutSeconds is the number of seconds that we will wait for a reply
+// from the server.
+static const int kTimeoutSeconds = 3;
+// kRetrialTimes is the number of time retrying to connect with server when if
+// it got failed
+static const int kRetrialTimes = 3;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+namespace roughtime {
+
+iotc_roughtime_state_t
+iotc_roughtime_create_socket(int* out_socket, const char* server_address) {
+  std::string address(server_address);
+
+  const size_t kColonOffset = address.rfind(':');
+  if (kColonOffset == std::string ::npos) {
+    iotc_debug_format(stderr, "No port number in server address: %s\n",
+                      address.c_str());
+    return IOTC_ROUGHTIME_ERROR;
+  }
+  std::string host(address.substr(0, kColonOffset));
+  const std::string kPortStr(address.substr(kColonOffset + 1));
+  int port = std::stoi(kPortStr.c_str());
```

Comment:
add `const`
+
this might fail, some error handling should be applied. Unit test coverage on false parameters?